src/openms/source/ANALYSIS/ID/SiriusMSConverter.cpp

```diff
@@ -188,6 +188,26 @@ namespace OpenMS
           count_to_neg = count_to_neg + 1;
         }
 
+        // set feature_charge value for msfile
+        if (p == IonSource::Polarity::POSITIVE && feature_charge == 1)
+        {
+          feature_charge = +1;
+        }
+        if (p == IonSource::Polarity::NEGATIVE && feature_charge == 1)
```

Comment:
the whole part could be written much shorter e.g.:
if (feature_charge == 0) { feature_charge = 1; }
if (p == IonSource::Polarity::NEGATIVE) { -abs(feature_charge) }

Reply:
and would already handle higher charge states

Reply:
alright will change that - thank you!

Reply:
> and would already handle higher charge states

yes, that would be good to have -- do not assume all charges are +1 or -1 ...

Reply:
That is correct. 
In general: SIRIUS is only able to use charge +1/-1  data